packages/core/src/routes/experience/classes/interaction-session.ts

```diff
@@ -0,0 +1,170 @@
+import { InteractionEvent, type VerificationType } from '@logto/schemas';
+import { z } from 'zod';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import { type WithLogContext } from '#src/middleware/koa-audit-log.js';
+import type TenantContext from '#src/tenants/TenantContext.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import type { Interaction } from '../type.js';
+
+import {
+  buildVerificationRecord,
+  verificationRecordDataGuard,
+  type VerificationRecord,
+} from './verifications/index.js';
+
+const interactionSessionResultGuard = z.object({
+  event: z.nativeEnum(InteractionEvent).optional(),
+  accountId: z.string().optional(),
+  profile: z.object({}).optional(),
+  verificationRecords: verificationRecordDataGuard.array().optional(),
+});
+
+/**
+ * InteractionSession
```

Comment:
"a short-lived session session"

Reply:
Classname and filename will be updated in a separate PR, to avoid unresolvable conflict with the following PRs.